static/skin/taskbar.js

```diff
@@ -76,9 +76,9 @@ document.addEventListener('DOMContentLoaded', function () {
         element: (item, data) => {
           let searchLink;
           if (data.value.kind == "path") {
-            searchLink = `${root}/${bookName}/${encodeURI(data.value.path)}`;
+            searchLink = `${root}/${bookName}/${encodeURIComponent(data.value.path)}`;
           } else {
-            searchLink = `${root}/search?content=${bookName}&pattern=${data.value.value}`;
+            searchLink = `${root}/search?content=${bookName}&pattern=${encodeURIComponent(data.value.value)}`;
```

Comment:
`data.value.path` and `data.value.value` must be `htmlDecode()`d before being URI-encoded.

Check with R&B on the Ray Charles ZIM.

1. The suggestion "Let's Go Get Stoned (R&B song)" points to http://localhost:8080/ray_charles/A%2FLet%26apos%3Bs_Go_Get_Stoned_(R%26amp%3BB_song) with a **Not Found** result

2. The search suggestion "containing 'R&B'..." points to http://localhost:8080/search?content=ray_charles&pattern=R%26amp%3BB%20 leading to No results were found for **"R&B "**

Reply:
This is fixed, but I would like to bring up another similar problem which is also present in old taskbar.
Searching for R$B, R#B doesn't bring up correct results
Old taskbar results (same with new) - 
![image](https://user-images.githubusercontent.com/26814569/181790146-ca396c05-51a0-4c0e-ae85-da9a302f514e.png)

Reply:
This issue (unlike the previous one) belongs to the search engine (Xapian). We shouldn't do anything in the front-end to handle it. Any workarounds must be implemented in the back-end.

Reply:
@juuz0 I woukd recommend to open a dedicated ticket, but not sure something can be done.